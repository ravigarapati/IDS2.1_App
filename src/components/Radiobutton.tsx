import React from 'react';
import {TouchableOpacity, View, StyleSheet} from 'react-native';
import CustomText from './CustomText';
import {Colors} from '../styles';

type RadiobuttonProps = {
  style?: any;
  checked: boolean;
  handleCheck?: any;
  text?: string;
  containerStyle?: any;
  accessibilityLabelText?: string;
  accessibilityHintText?: string;
  disabled?: boolean;
  testID?: string;
};

const Radiobutton = ({
  checked,
  handleCheck,
  text,
  style,
  containerStyle,
  accessibilityLabelText = '',
  accessibilityHintText = '',
  disabled = false,
  testID = '',
}: RadiobuttonProps) => {
  return (
    <View
      accessible={true}
      accessibilityLabel={
        accessibilityLabelText ? accessibilityLabelText : undefined
      }
      accessibilityHint={
        accessibilityHintText ? accessibilityHintText : undefined
      }
      onAccessibilityTap={() => handleCheck()}
      style={[{flexDirection: 'row'}, containerStyle]}>
      <TouchableOpacity
        testID={testID}
        disabled={disabled}
        onPress={() => {
          handleCheck();
        }}>
        <View
          style={[
            styles.radioButton,
            {
              backgroundColor: disabled
                ? Colors.grayDisabled
                : checked
                ? 'rgba(0, 73, 117, 1)'
                : 'rgba(164, 171, 179, 1)',
            },
          ]}>
          {checked ? <View style={styles.checkedRadioButton} /> : null}
        </View>
      </TouchableOpacity>
      <CustomText
        size={16}
        text={`  ${text}`}
        allowFontScaling={true}
        style={style}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  radioButton: {
    height: 24,
    width: 24,
    borderRadius: 12,
    alignItems: 'center',
    justifyContent: 'center',
  },
  checkedRadioButton: {
    height: 15,
    width: 15,
    borderRadius: 12,
    borderColor: 'blue',
    backgroundColor: 'white',
  },
});

export default Radiobutton;
