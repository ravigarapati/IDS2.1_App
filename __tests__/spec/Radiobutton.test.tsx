import 'react-native';
import React from 'react';
import Radiobutton from '../../src/components/Radiobutton';
import renderer from 'react-test-renderer';

describe('Radiobutton Component', () => {
  test('Component to be defined', () => {
    const tree = renderer
      .create(<Radiobutton checked={true} handleCheck={() => {}} text="" />)
      .toJSON();
    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const tree = renderer
      .create(<Radiobutton checked={true} handleCheck={() => {}} text="" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
