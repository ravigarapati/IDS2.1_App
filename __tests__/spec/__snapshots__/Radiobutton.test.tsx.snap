// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radiobutton Component Component to be rendered 1`] = `
<View
  accessible={true}
  onAccessibilityTap={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID=""
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 12,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
          Object {
            "backgroundColor": "rgba(0, 73, 117, 1)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "blue",
            "borderRadius": 12,
            "height": 15,
            "width": 15,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    numberOfLines={0}
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#000000",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "fontFamily": "BoschSans-Regular",
        },
        Object {},
      ]
    }
  >
      
  </Text>
</View>
`;
